aptos-move/block-executor/src/captured_reads.rs

```diff
@@ -583,6 +584,30 @@ impl<T: Transaction> CapturedReads<T> {
         })
     }
 
+    #[allow(dead_code)]
+    pub(crate) fn validate_module_reads(
+        &self,
+        // TODO(George): Add concrete or generic type for module storage entry instead of unit.
+        module_storage: &VersionedModuleStorage<T::Key, ()>,
+        idx_to_validate: TxnIndex,
+    ) -> bool {
+        if self.speculative_failure {
+            return false;
+        }
+
+        self.module_reads.iter().all(|k| {
+            use ModuleReadError::*;
+
+            match module_storage.read(k, idx_to_validate) {
+                Ok(_) => {
+                    // TODO(George): Use hash or transaction index to distinguish the reads.
+                    true
+                },
+                Err(Dependency(_)) | Err(Uninitialized) => false,
```

Comment:
I wonder (even for the data), is it possible to have Uninitialized during validation? maybe it makes sense to use the invariant that validation requires a read to exist, and the read would have initialized the entry? I don't mean adding an unreachable immediately, but maybe start with an error!() @igor-aptos what do you think?